pulp_python/tests/upgrade/post/test_publish.py

```diff
@@ -0,0 +1,131 @@
+        remote = self.remote_api.create(body)
+
+        repo = self.repo_api.create(gen_repo())
+
+        repository_sync_data = RepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        self.repo = self.repo_api.read(repo.pulp_href)
+
+    def test_all(self):
+        """Test whether a particular repository version can be published.
+
+        1. Create a repository with at least 2 repository versions.
+        2. Create a publication by supplying the latest ``repository_version``.
+        3. Assert that the publication ``repository_version`` attribute points
+           to the latest repository version.
+        4. Create a publication by supplying the non-latest ``repository_version``.
+        5. Create distribution.
+        6. Assert that the publication ``repository_version`` attribute points
+           to the supplied repository version.
+        7. Assert that an exception is raised when providing two different
+           repository versions to be published at same time.
+        """
+        # Step 1
+        repo_content = get_content(self.repo.to_dict())[PYTHON_CONTENT_NAME]
+        print(repo_content)
+        for file_content in repo_content:
+            repository_modify_data = RepositoryAddRemoveContent(
+                remove_content_units=[file_content["pulp_href"]]
+            )
+            modify_response = self.repo_api.modify(self.repo.pulp_href, repository_modify_data)
+            monitor_task(modify_response.task)
+        version_hrefs = tuple(ver["pulp_href"] for ver in get_versions(self.repo.to_dict()))
+        print(version_hrefs)
+        non_latest = choice(version_hrefs[:-1])
+
+        # Step 2
+        publish_data = PythonPythonPublication(repository=self.repo.pulp_href)
+        publication = self.create_publication(publish_data)
+
+        # Step 3
+        self.assertEqual(publication.repository_version, version_hrefs[-1])
+
+        # Step 4
+        publish_data = PythonPythonPublication(repository_version=non_latest)
+        publication = self.create_publication(publish_data)
+
+        # Step 5
+        body = gen_distribution()
+        body["base_path"] = "pulp_post_upgrade_test"
+        body["publication"] = publication.pulp_href
+
+        distribution_response = self.distributions.create(body)
+        created_resources = monitor_task(distribution_response.task).created_resources
+        distribution = self.distributions.read(created_resources[0])
+
+        # Step 6
+        self.assertEqual(publication.repository_version, non_latest)
+
+        # Step 7
+        with self.assertRaises(ApiException):
+            body = {"repository": self.repo.pulp_href, "repository_version": non_latest}
+            self.publications.create(body)
+
+        # Step 8
+        self.assertIn("/pypi/pulp_post_upgrade_test/", distribution.base_url)
+
```

Comment:
If these tests only run in the CI then the base url should be `https://pulp:80/pypi/pulp_post_upgrade_test/` I think. It's fine this way too. 

Also, I think the publication & distribution from the pre-test aren't destroyed correct? If not then should a test be added here to ensure you can access them from the new endpoint location?

Reply:
I think I can use the new endpoint when syncing, I'll change that